zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/BackupManager.java

```diff
@@ -623,22 +634,31 @@ protected void backupComplete(BackupFile file) throws IOException {
    * @param tmpDir temporary directory
    * @param backupIntervalInMinutes the interval backups should run at in minutes
    */
-  public BackupManager(File snapDir, File dataLogDir, File backupStatusDir, File tmpDir, int backupIntervalInMinutes,
-      BackupStorageProvider backupStorageProvider) throws IOException {
+  public BackupManager(File snapDir, File dataLogDir, File backupStatusDir, File tmpDir,
+      int backupIntervalInMinutes, BackupStorageProvider backupStorageProvider) throws IOException {
+    this(snapDir, dataLogDir, backupStatusDir, tmpDir, backupIntervalInMinutes,
```

Comment:
Is this constructor necessary anymore? If you want to leave this for testing, I think we should remove this constructor, and modify tests so that they pass in empty/null values where appropriate.

Reply:
I left it here for backward compatibility. But good point, the only place we are using it is in tests, so we can remove it neatly. I'll remove the constructor and modify the tests.